pkg/chunked/compression_linux.go

```diff
@@ -160,9 +160,9 @@ func readEstargzChunkedManifest(blobStream ImageSourceSeekable, blobSize int64,
 }
 
 func openTmpFile(tmpDir string) (*os.File, error) {
-	file, err := os.OpenFile(tmpDir, unix.O_TMPFILE|unix.O_RDWR|unix.O_CLOEXEC|unix.O_EXCL, 0o600)
+	fd, err := unix.Open(tmpDir, unix.O_TMPFILE|unix.O_RDWR|unix.O_CLOEXEC|unix.O_EXCL, 0o600)
```

Comment:
Back when reviewing the original commit I checked that
 - os.OpenFile flags argument type is `int` (i.e. a generic one);
 - it is passed almost as is onto syscall.Open ("almost" since O_LARGEFILE and O_CLOEXEC are added);
 - `os.O_*`, `syscall.O_*`, and `unix.O_*` flag values should be equal (although there is no guarantee);
 - a lot of existing code uses `unix.O_*` flags with `os.OpenFile`;
 
and so I guesses it's OK as it is.

With this change, though
 - we lose the ability to retry on EINTR;
 - as far as I understand, `os.OpenFile` adds the file to the internal poller, while the new code doesn't (which probably doesn't matter here).
 
 All in all, I slightly prefer the old way.